src/trainer_see.c

```diff
@@ -369,14 +369,26 @@ bool8 CheckForTrainersWantingBattle(void)
 
     for (i = 0; i < OBJECT_EVENTS_COUNT; i++)
     {
+        for (u8 j = 0; j < OBJECT_EVENTS_COUNT; j++)
+        {
+            k = 0;
+            if (gObjectEvents[j].localId == i)
```

Comment:
`i` is used for `localId` here, isn't it?

But I don't think it matters, because with a sort you can simply loop through just the object events that are in `gObjectEvents` and sort on their local IDs at the end, rather than having to loop through all the possible local IDs :)